internal/compiler/v2/compiler_v2.go

```diff
@@ -117,7 +116,7 @@ func (c *CompilerV2) HandleDynamicVar(v taskfile.Var, _ string) (string, error)
 	result := strings.TrimSuffix(stdout.String(), "\n")
 
 	c.dynamicCache[v.Sh] = result
-	c.Logger.VerboseErrf(logger.Magenta, `task: dynamic variable: '%s' result: '%s'`, v.Sh, result)
+	log.VerboseErrf(log.Magenta, "task: dynamic variable: %q result: %q\n", v.Sh, result)
```

Comment:
Logger should handle trailing whitespace

Reply:
format loggers shouldn't handle trailing whitespace - In the same way that `fmt.Printf` does not. This gives the caller more control of the output. We should be consistent with the standard library.

Reply:
Stdlib log does not require trailing \n, fmt does; structured logging also does not require \n (in fact, it is escaped to prevent log forgery, prefer %q for params in stdlib).